src/pkg/services/m365/api/mail.go

```diff
@@ -165,6 +165,55 @@ func (c Mail) GetContainerByID(
 	return c.GetFolder(ctx, userID, containerID)
 }
 
+// GetContainerByName fetches a folder by name
+func (c Mail) GetContainerByName(
+	ctx context.Context,
+	userID, containerName string,
+) (graph.Container, error) {
+	filter := fmt.Sprintf("displayName eq '%s'", containerName)
+	options := &users.ItemMailFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemMailFoldersRequestBuilderGetQueryParameters{
+			Filter: &filter,
+		},
+	}
+
+	ctx = clues.Add(ctx, "container_name", containerName)
+
+	resp, err := c.Stable.
+		Client().
+		Users().
+		ByUserId(userID).
+		MailFolders().
+		Get(ctx, options)
```

Comment:
is this a recursive enumeration of all mail folders? Is it meant to be recursive?

Reply:
It is not recursive.  That works for now because our restores are strictly top-level folders.  I'll add a TODO that the later implementation will need to expand on that.  I don't think we'll want a recursive query at that time either.  But mailFolders within a given parent ID will work.

Reply:
> That works for now because our restores are strictly top-level folders

I don't think this statement fits with how this function is being used. Looking at the code, this function is called in `getOrPopulateContainer` which is called for _any_ container we make in `createDestination`, whether or not it's a top-level container. This will work for events (and probably for contacts) because for those we don't expect a full hierarchy. However, it's quite possible it will result in an incorrect hierarchy for mail because it may end up selecting the wrong folder (or no folder at all), especially if it's supposed to be looking for a nested folder

Reply:
oh hell, thanks, good catch.

Reply:
Okay, this is now fixed.  The interface takes an additional "parentContainerID", so we'll always try to find folder names within a specific container.